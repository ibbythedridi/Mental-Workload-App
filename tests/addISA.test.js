import React from 'react';
import renderer from 'react-test-renderer';
import AddISA from '../screens/addISA';

it('renders correctly', () => {
    const tree = renderer.create(<AddISA />).toJSON();
    expect(tree).toMatchSnapshot();
});

// This checks the correct component(s) are existent in the JSON (views and buttons)
describe("<AddISA />", () => {
  it('has 5 children', () => {
    const tree = renderer.create(<AddISA />).toJSON();
    expect(tree.children.length).toBe(5);
  });
});