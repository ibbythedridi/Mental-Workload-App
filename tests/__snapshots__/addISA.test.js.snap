// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 10,
    }
  }
>
  <Text>
    Date
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="DD/MM/YYYY"
    rejectResponderTermination={true}
    style={
      Object {
        "borderWidth": 1,
        "height": 40,
        "marginBottom": 10,
        "padding": 5,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <Text>
    Time
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="hh:mm:ss"
    rejectResponderTermination={true}
    style={
      Object {
        "borderWidth": 1,
        "height": 40,
        "marginBottom": 10,
        "padding": 5,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <Text>
    Workload Rating
  </Text>
  <TextInput
    allowFontScaling={true}
    keyboardType="number-pad"
    onChangeText={[Function]}
    placeholder="0-5"
    rejectResponderTermination={true}
    style={
      Object {
        "borderWidth": 1,
        "height": 40,
        "marginBottom": 10,
        "padding": 5,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <Text>
    Summary
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Summary of task"
    rejectResponderTermination={true}
    style={
      Object {
        "borderWidth": 1,
        "height": 40,
        "marginBottom": 10,
        "padding": 5,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#5863F8",
          "borderRadius": 7,
          "marginVertical": 5,
          "paddingHorizontal": 10,
          "paddingVertical": 10,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "center",
            "textTransform": "uppercase",
          }
        }
      >
        Submit
      </Text>
    </View>
  </View>
</View>
`;
